### PR TITLE
Change sender worker to send a set of requests concurrently

### PR DESCRIPTION
**Pull request by justinvdm**
Since the kind of requests that will be given to the retry api are often going to be requests that timed out (and are likely to time out again), it doesn't make sense to block on each request. It would be more useful to have a set of requests running concurrently.

### CODE REVIEW

**Comment by justinvdm** [comment]
Depends on #25, otherwise ready for review.

File: vumi_http_retry/workers/sender/worker.py [hunk 1]
```diff
@@ -93,19 +97,26 @@ def retry(self, req):
...
                 break
 
-            yield self.retry(req)
+            yield limiter.add(self.retry, req)
+
+        yield limiter.done()
 
     def on_error(self, err):
         log.err(err)
+        reactor.stop()
```
Comment (hodgestar) [inline vumi_http_retry/workers/sender/worker.py:119]: Seems unrelated to the PR but +1 on this.

**Comment by hodgestar** [comment]
Spotted one minor typo, otherwise looks good.

File: vumi_http_retry/workers/sender/worker.py (hunk 1 above)
Comment (justinvdm) [inline vumi_http_retry/workers/sender/worker.py:119]: Woops, sorry for unrelated change, I think I was seeing how this affects tests and forgot to remove.

**Comment by justinvdm** [comment]
Ready for re-review.

**Comment by hodgestar** [comment]
:+1:
